primitives/consensus/slots/src/lib.rs

```diff
@@ -94,6 +94,30 @@ impl From<Slot> for u64 {
 	}
 }
 
+/// A slot duration defined in milliseconds.
+#[derive(Clone, Copy, Debug, Encode, Decode, Hash, PartialOrd, Ord, PartialEq, Eq, TypeInfo)]
+pub struct SlotDuration(u64);
+
+impl SlotDuration {
+	/// Initialize from the given milliseconds.
+	pub fn from_millis(millis: u64) -> Self {
+		SlotDuration(millis)
+	}
+}
+
+#[cfg(feature = "std")]
+impl SlotDuration {
+	/// Returns `self` as [`std::time::Duration`].
+	pub fn as_duration(&self) -> std::time::Duration {
+		std::time::Duration::from_millis(self.0)
+	}
+
+	/// Returns `self` as a `u128` representing the duration in milliseconds.
+	pub fn as_millis(&self) -> u128 {
```

Comment:
I know why you did it (because `Duration::as_millis` returns a `u128`), but it's kinda strange that `from_millis` takes a `u64` but `as_millis` returns a `u128`. Personally I'd probably have it return `u64` here to be consistent with `from_millis` and just do a manual cast when it needs to be used as a divisor of the `Duration`.